lib/ex_aws/s3.ex

```diff
@@ -511,35 +511,30 @@ defmodule ExAws.S3 do
   ```
   enumerable = ["hello world"] |> Stream.with_index(1)
 
-  Flow.new(stages: 4, max_demand: 2)
-  |> Flow.from_enumerable(enumerable)
+  Flow.from_enumerable(enumerable, stages: 4, max_demand: 2)
   |> S3.upload("my-bucket", "test.txt")
   |> ExAws.request! #=> :done
   ```
+
+  ## Options
+
+  These options are specific to this function
+
+  * `:max_concurrency` -- The number of concurrent processes reading from this
+     stream. Only applies when uploading a stream.
+
+  All other options (ex. `:content_type`) are passed through to
+  `ExAws.S3.initiate_multipart_upload/3`.
```

Comment:
I added documentation for the options, but didn't add it to the typespec because I wasn't sure how to easily include all of the options that are passed through to `ExAws.S3.initiate_multipart_upload/3`